### ISSUE TITLE
Increase contrast of checkout error messages when using dark mode

### ISSUE THREAD

**Issue by nielslange** [woocommerce/woocommerce-gutenberg-products-block#5740]
## Describe the bug

Currently, the color contrast of the form labels and the error messages in the checkout block, when using the dark mode, is insufficient, when an error occurs.

## To reproduce

1. Use a theme that supports dark mode, e.g. [Eksell](https://wordpress.org/themes/eksell/).
2. Create a test page and add the checkout block.
3. Click on the checkout block and activate the option `Style » Dark mode inputs` in the sidebar.
4. Save that test page and go to the frontend.
5. Add a product to the cart and go to the test page with the checkout block.
6. Clear all form fields, if they're pre-filled, and click on place order.
7. See the error messages with insufficient color contrast.
  
## Current behavior

The color contrasts of the labels and error messages when using dark mode are:

- **2.72:1** for the label within the form field (text: `#CC1818`, background: `#10282f`)
- **2.53:1** for the error message below the form field (text: `#CC1818`, background: `#192d33`)

This makes the labels and error messages very hard to read.

## Expected behavior

https://wordpress.org/about/accessibility/ states:

> WordPress aims to make the WordPress Admin and bundled themes fully WCAG 2.0 AA compliant where possible.

https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html states:

> Success Criterion (Level AA): The visual presentation of text and images of text has **a contrast ratio of at least 4.5:1**, [...]

With WCAG 2.0 AA in mind, the color contrast needs to be increased at least to 4.5:1. One of various tools to find a sufficient color contrast is https://color.adobe.com/create/color-contrast-analyzer.

## Screenshots

<table>
<tr>
<td valign="top">Checkout block:
<br><br>

![error-message-contrast](https://user-images.githubusercontent.com/3323310/152970940-a316b9ad-a505-43b1-886e-f132ebbabe4d.png)
</td>
<td valign="top">Contrast of labels:
<br><br>

![label](https://user-images.githubusercontent.com/3323310/152972066-7d0919ee-b70a-482e-bf2a-6dc508949202.png)
</td>
<td valign="top">Contrast of error messages:
<br><br>

![error-message](https://user-images.githubusercontent.com/3323310/152972048-6001e018-3d7b-4b32-84f3-661a9c8b6b68.png)
</td>
</tr>
</table>

**Comment by wiksien** [comment]
I will do that.

**Comment by nielslange** [comment]
Hi @wiksien 👋

Thanks for taking care of this! Please let me know if you have any questions. 😀

**Comment by wiksien** [comment]
@nielslange When I look at the code, I can see that many elements besides just the text have a color of #CC1818. Should I change every instance of this colour that happens to exist in code, or just the ones that are text?

**Comment by nielslange** [comment]
> @nielslange When I look at the code, I can see that many elements besides just the text have a color of #CC1818. Should I change every instance of this colour that happens to exist in code, or just the ones that are text?

Please only change the color of the elements as mentioned above, and keep in mind to only change them when `dark mode inputs` are activated. If the checkout block is not shown with `dark mode inputs`, the color should remain `#cc1818`.

When the checkout block is using `dark mode`, the checkout page gets the additional body class `has-dark-mode-palette`. You could use this class to target the desired elements. The current SCSS code for the error color looks like this:

```css
.wc-block-components-validation-error {
	@include font-size(smaller);
	color: $alert-red;
	max-width: 100%;
	white-space: normal;
}
```

Using `has-dark-mode-palette`, you could add something like this:

```css
.has-dark-mode-palette .wc-block-components-validation-error {
	color: lighten($alert-red, 10%);
}
```

I hope this helps to proceed with this issue.

**Comment by wiksien** [comment]
Unfortunately, due to personal issues I won't be able to fix this issue in a reasonable amout of time (probably will come back to it in a couple of days). For now, if someone else wants to do it, go for it. Thank you nielslange for guidance.

**Comment by nielslange** [comment]
Thanks for letting me know, @wiksien.
